File: src/algorithms/get-min-distance.js
/**
 * Given an integer array nums and two integers target and start, finds an index
 * such that nums[i] === target and Math.abs(i - start) is minimized.
 * 
 * Returns Math.abs(i - start);
 * 
 * @param {number[]} nums 
 * @param {*} target 
 * @param {*} start 
 * @returns 
 */
function getMinDistance (nums, target, start) {
  let minDistance = Math.max(...nums);
  let index = minDistance;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === target) {
      index = i;
      minDistance = Math.min(minDistance, Math.abs(index - start));
    }
  }
  return minDistance;
}

File: src/algorithms/get-min-distance.py
def getMinDistance(nums, target, start):
    """Given an integer array nums (0-indexed) and two integers target and 
    start, find an index i such that nums[i] == target and abs(i - start) is 
    minimized. Note that abs(x) is the absolute value of x. Return abs(i - start). It is guaranteed that target exists in nums.

    Args:
        nums (list[int]): A list of integers.
        target (int): A target integer.
        start (int): A starting index.

    Returns:
        int: Returns the absolute distance between target number and start.
    """
    res = len(nums)
    for i in range(len(nums)):
        if nums[i] == target:
            res = min(abs(i - start), res)
    return res
